folly/portability/Windows.h

```diff
@@ -37,6 +37,7 @@
 #include <direct.h> // nolint
 #endif
 
+#define NOMINMAX
```

Comment:
Ah, yep, sorry, I've been mostly out of the office for the last week. The right fix is definitely to add some of the defines here, but in order to do that, an extra check needs to be added to ensure that `Windows.h` and winsock haven't been included before this header is included.